### PR TITLE
Revert "Revert "Create Talk rooms for appointments""

### PR DESCRIPTION
**Pull request by ChristophWurst**
Reverts nextcloud/calendar#5306 and brings back https://github.com/nextcloud/calendar/pull/4025

- [x] DCO
- [x] Rename migration
- [x] Revert https://github.com/nextcloud/calendar/pull/5312

## How to test

1) Install and enable Calendar and Talk
2) Create an appointment config in calendar, tick the *Create talk room* checkbox
3) Open the booking URL in a private tab
4) Book and confirm an appointment
5) Open Talk as the organizer
6) See the Talk conversation created for the event

### CODE REVIEW

**Review by hamza221 — APPROVED** [review]
image
2 talk rooms are created but it's a problem with the original Pr I'm guessing. I can try to solve it when this is merged.
Also The booking page needs  a loading animation, I'll create an issue 
But aside from this works perfectly 👍

**Comment by ChristophWurst** [comment]
Did you book two appointments? The room will not be shared. If person a and person b book, they get two separate rooms. If person a books two slots, there will still be two rooms.

**Comment by hamza221** [comment]
No just one appointment with one user

**Comment by hamza221** [comment]
image
Looks deterministic

**Comment by ChristophWurst** [comment]
I'll test another time. This doesn't seem right

**Comment by hamza221** [comment]
The user receives only one confirmation email, with the first room's link .

**Comment by st3iny** [comment]
The `AppointmentBookedEvent` is emitted twice in `\OCA\Calendar\Service\Appointments\BookingService::confirmBooking`.

File: lib/Service/Appointments/BookingService.php [hunk 1]
```diff
@@ -111,10 +118,26 @@ public function confirmBooking(Booking $booking, AppointmentConfig $config): Boo
...
 		}
 
-		$calendar = $this->calendarWriter->write($config, $startObj, $booking->getDisplayName(), $booking->getEmail(), $booking->getDescription());
+		// TODO: inject broker here and remove indirection
+		$this->eventDispatcher->dispatchTyped(
+			new AppointmentBookedEvent(
+				$booking,
+				$config,
+			)
+		);
```
Comment (st3iny) [inline lib/Service/Appointments/BookingService.php:127]: :boom:

File: lib/Service/Appointments/BookingService.php [hunk 2]
```diff
@@ -128,6 +151,14 @@ public function confirmBooking(Booking $booking, AppointmentConfig $config): Boo
 			$this->logger->warning('Could not send booking information notification after confirmation by user ' . $booking->getEmail(), ['exception' => $e]);
 		}
 
+
+		$this->eventDispatcher->dispatchTyped(
+			new AppointmentBookedEvent(
+				$booking,
+				$config
+			)
+		);
```
Comment (st3iny) [inline lib/Service/Appointments/BookingService.php:160]: :boom: or this one

Comment (ChristophWurst) [reply lib/Service/Appointments/BookingService.php]: this because the listener sets the talk URL on the object passed through the event

**Review by kesselb — APPROVED** [review]

**Pull request merged**